projects/aws/image-builder/builder/utils.go

```diff
@@ -76,6 +117,48 @@ func GetSupportedReleaseBranches() []string {
 	return supportReleaseBranches
 }
 
+func getEksDReleaseBranchesWithNumber() (map[string]string, error) {
+	buildToolingPath, err := getRepoRoot()
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	eksDReleaseBranchesFile := filepath.Join(buildToolingPath, "EKSD_LATEST_RELEASES")
+	eksDReleaseBranchesFileData, err := os.ReadFile(eksDReleaseBranchesFile)
+
+	eksDReleaseBranchesWithNumber := make(map[string]string)
+	var eksDReleaseBranchesDataWithNumber EksDReleases
+	err = yaml.Unmarshal(eksDReleaseBranchesFileData, &eksDReleaseBranchesDataWithNumber)
```

Comment:
Are you using `sigs.k8s.io/yaml` here for unmarshalling? I have heard that this module works best for unmarshalling Kubernetes objects and for arbitrary YAML, we can just use `github.com/ghodss/yaml`. This is what we do in EKS-A release too ([1], [2])

[[1]](https://github.com/aws/eks-anywhere/blob/main/release/pkg/filereader/file_reader.go#L28)
[[2]](https://github.com/aws/eks-anywhere/blob/main/release/pkg/filereader/file_reader.go#L107)